x-pack/plugins/security_solution/public/management/pages/response_actions/view/response_actions_list.tsx

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CriteriaWithPagination,
+  EuiAvatar,
+  EuiBadge,
+  EuiBasicTable,
+  EuiButtonIcon,
+  EuiDescriptionList,
+  EuiEmptyPrompt,
+  EuiFacetButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiScreenReaderOnly,
+  EuiText,
+  EuiToolTip,
+  RIGHT_ALIGNMENT,
+} from '@elastic/eui';
+
+import type { HorizontalAlignment } from '@elastic/eui';
+import React, { memo, useCallback, useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import styled from 'styled-components';
+import { FormattedDate } from '../../../../common/components/formatted_date';
+import {
+  ActionDetails,
+  LogsEndpointActionResponse,
+  EndpointActivityLogAction,
+} from '../../../../../common/endpoint/types';
+import type { EndpointActionListRequestQuery } from '../../../../../common/endpoint/schema/actions';
+import { AdministrationListPage } from '../../../components/administration_list_page';
+import { ManagementEmptyStateWrapper } from '../../../components/management_empty_state_wrapper';
+import { useGetEndpointActionList } from '../../../hooks';
+import { OUTPUT_MESSAGES, TABLE_COLUMN_NAMES, UX_MESSAGES } from '../translations';
+import { MANAGEMENT_PAGE_SIZE_OPTIONS } from '../../../common/constants';
+import { useTestIdGenerator } from '../../../hooks/use_test_id_generator';
+
+// Truncated usernames
+const StyledFacetButton = styled(EuiFacetButton)`
+  .euiText {
+    margin-top: 0.38rem;
+    overflow-y: visible !important;
+  }
+`;
+
+export const ResponseActionsList = memo<
+  EndpointActionListRequestQuery & { hideHeader?: boolean; hideHostNameColumn?: boolean }
+>(({ agentIds, commands, userIds, hideHeader = false, hideHostNameColumn = false }) => {
+  const getTestId = useTestIdGenerator('response-actions-list');
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<{
+    [k: ActionDetails['id']]: React.ReactNode;
+  }>({});
+
+  const [queryParams, setQueryParams] = useState<EndpointActionListRequestQuery>({
+    page: 0,
```

Comment:
Why is page `0`?  It should be `1` based in the API.

Reply:
I'm keep forgetting if we decided to stick to 0 or 1-based paging. 😅 In the API it is 0 based as it's using the management_default_page const.

Reply:
a8a0eb693fe3885fbf3089bfe9b90afcda46e75d